jetty-fcgi/fcgi-server/src/main/java/org/eclipse/jetty/fcgi/server/HttpChannelOverFCGI.java

```diff
@@ -260,13 +263,45 @@ private boolean doOnIdleTimeout(Throwable x)
     @Override
     public void recycle()
     {
-        try (AutoLock l = _lock.lock())
+        if (!_contentQueue.isEmpty())
+            throw new AssertionError("unconsumed content: " + _contentQueue);
+        _specialContent = null;
+        super.recycle();
```

Comment:
`super.recycle()` should be the first call, otherwise there is a risk that demanding content could be left pending. Of course, this bug does not exist for FCGI and HTTP/1 but that was a real HTTP/2 bug so I'd prefer that we keep this pattern in sync across all protocols.

Reply:
Done. Will not change HTTP/1.1 though.